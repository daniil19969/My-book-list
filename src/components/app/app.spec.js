import React from "react";
import App from "./app";

const componentDidMountSpy = jest.spyOn(App.prototype, "componentDidMount");

const setUp = () => shallow(<App />);
describe("App component", () => {

  let component;
  beforeEach(() => {
    jest.spyOn(window, "addEventListener");
    jest.spyOn(window, "removeEventListener");
    component = setUp();
  });

  afterEach(() => {
    window.addEventListener.mockRestore();
    window.removeEventListener.mockRestore();
  });




  it("should render component App", () => {
    expect(component).toMatchSnapshot();
  });

  describe("Lifecycle method", () => {

    it("should call componentDidMount", () => {
      expect(componentDidMountSpy).toHaveBeenCalledTimes(2);
    });
  });

});
