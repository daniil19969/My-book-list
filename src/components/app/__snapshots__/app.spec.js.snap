// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render component App 1`] = `
<div>
  <Header />
  <table
    className="table"
  >
    <thead
      className="head"
    >
      <tr>
        <th
          rowSpan="2"
        >
          Обложка
        </th>
        <th>
          Название
        </th>
        <th>
          Автор
        </th>
        <th>
          Жанр
        </th>
      </tr>
      <tr>
        <SearchTitle
          onSearchChange={[Function]}
        />
        <SearchAuthor
          onSearchChange={[Function]}
        />
        <SearchGenre
          onSearchChange={[Function]}
        />
      </tr>
    </thead>
    <tbody>
      <BookList
        book={Array []}
      />
    </tbody>
  </table>
</div>
`;
